### PR TITLE
armagetronad: 0.2.8.3.5 -> 0.2.9.1.0

### PR DESCRIPTION
**Pull request by numinit**
###### Motivation for this change

Updating armagetronad to latest and adding dedicated server with different compile flags.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/armagetronad/default.nix [hunk 1]
```diff
@@ -1,30 +1,47 @@
...
-  buildInputs = [ SDL SDL_image libxml2 libjpeg libpng libGLU libGL zlib ];
+  configureFlags = [
+    "--enable-memmanager"
+    "--enable-automakedefaults"
+    "--disable-useradd"
+    "--disable-initscripts"
+    "--disable-etc"
+    "--disable-uninstall"
+    "--disable-sysinstall"
+  ] ++ (lib.optional dedicatedServer "--enable-dedicated");
```
Comment (SuperSandro2000) [inline pkgs/games/armagetronad/default.nix:33]:
```suggestion
  ] ++ lib.optional dedicatedServer "--enable-dedicated";
```

File: pkgs/games/armagetronad/default.nix [hunk 2]
```diff
@@ -1,30 +1,47 @@
...
+
+  nativeBuildInputs = [ pkgconfig ];
```
Comment (SuperSandro2000) [inline pkgs/games/armagetronad/default.nix:35]:
```suggestion
  nativeBuildInputs = [ pkg-config ];
```

File: pkgs/games/armagetronad/default.nix [hunk 3]
```diff
@@ -1,30 +1,47 @@
-{ lib, stdenv, fetchurl, SDL, libxml2, SDL_image, libjpeg, libpng, libGLU, libGL, zlib }:
+{ dedicatedServer ? false
+, lib, stdenv, fetchurl
+, pkgconfig, SDL, libxml2, SDL_image, libjpeg, libpng, libGLU, libGL, zlib }:
```
Comment (SuperSandro2000) [inline pkgs/games/armagetronad/default.nix:3]:
```suggestion
, pkg-config, SDL, libxml2, SDL_image, libjpeg, libpng, libGLU, libGL, zlib
, dedicatedServer ? false }:
```

File: pkgs/games/armagetronad/default.nix [hunk 4]
```diff
@@ -1,30 +1,47 @@
-{ lib, stdenv, fetchurl, SDL, libxml2, SDL_image, libjpeg, libpng, libGLU, libGL, zlib }:
+{ dedicatedServer ? false
+, lib, stdenv, fetchurl
```
Comment (SuperSandro2000) [inline pkgs/games/armagetronad/default.nix:2]:
```suggestion
{ lib, stdenv, fetchurl
```

File: pkgs/games/armagetronad/default.nix [hunk 5]
```diff
@@ -1,30 +1,47 @@
...
+    "--disable-initscripts"
+    "--disable-etc"
+    "--disable-uninstall"
+    "--disable-sysinstall"
+  ] ++ (lib.optional dedicatedServer "--enable-dedicated");
+
+  nativeBuildInputs = [ pkgconfig ];
+
+  buildInputs = [ libxml2 zlib ] ++
+    (lib.optionals (!dedicatedServer) [ SDL SDL_image libxml2 libjpeg libpng libGLU libGL ]);
```
Comment (SuperSandro2000) [inline pkgs/games/armagetronad/default.nix:38]:
```suggestion
  buildInputs = [ libxml2 zlib ]
    ++ lib.optionals (!dedicatedServer) [ SDL SDL_image libxml2 libjpeg libpng libGLU libGL ];
```

File: pkgs/top-level/all-packages.nix [hunk 6]
```diff
@@ -26248,6 +26248,8 @@ in
 
   armagetronad = callPackage ../games/armagetronad { };
 
+  armagetronad-dedicated = callPackage ../games/armagetronad { dedicatedServer = true; };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:26251]:
```suggestion
  armagetronad-server = callPackage ../games/armagetronad { dedicatedServer = true; };
```

or dedicated server which is a bit long.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix (hunk 6 above)
Comment (numinit) [reply pkgs/top-level/all-packages.nix:26251]: Hmm, the compiled binary is called armagetronad-dedicated. I was going for consistency with that.

Comment (SuperSandro2000) [reply pkgs/top-level/all-packages.nix:26251]: Then it makes sense to keep the name.

File: pkgs/games/armagetronad/default.nix [hunk 7]
```diff
@@ -1,30 +1,47 @@
...
+
+  buildInputs = [ libxml2 zlib ]
+    ++ lib.optionals (!dedicatedServer) [ SDL SDL_image libxml2 libjpeg libpng libGLU libGL ];
 
   meta = with lib; {
     homepage = "http://armagetronad.org";
-    description = "An multiplayer networked arcade racing game in 3D similar to Tron";
+    description = "A multiplayer networked arcade racing game in 3D similar to Tron";
+    maintainers = with maintainers; [ numinit ];
     license = licenses.gpl2;
```
Comment (SuperSandro2000) [inline pkgs/games/armagetronad/default.nix:44]: gpl2Plus or gpl2Only?

**Comment by SuperSandro2000** [comment]
Can you please split the commit into ``armagetronad: 0.2.8.3.5 -> 0.2.9.1.0`` and ``armagetronad-dedicated: init at 0.2.9.1.0``.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 111273` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>armagetronad</li>
<li>armagetronad-dedicated</li>
</ul>
</details>

**Comment by numinit** [comment]
Done.

**Pull request merged**